### ISSUE TITLE
Stepper component

### ISSUE THREAD

**Issue by adamsilver** [adamsilver/f#118]
```
<div class="stepper">
  <button type="button" aria-label="Decrement people aged 16 and over">&minus;</button>
  <input type="number" id="adults" name="adults">
  <button type="button" aria-label="Increment people aged 16 and over">&plus;</button>
  <div class="vh" role="status" aria-live=""></div>
</div>
```
H: 
I'm unsure about this setup.
- If you use the How many... text as a group label (via <legend>) then I think the verbose aria-label values are unnecessary. Could just be "decrement" and "increment".
- The aria-live region is overkill. You could just whack the live region attributes directly on the input. This doesn't work everwhere, but is only an enhancement anyway. The user knows what they are doing from the context and button labels.

Suggestion:
```
<fieldset>
    <legend id="under-two">How many infants, under two years old, are traveling?</legend>
    <button type="button" aria-label="Decrement">&minus;</button>
    <input type="number" id="adults" name="adults" aria-labelledby="under-two">
    <button type="button" aria-label="Increment">&plus;</button>
</fieldset>
```
(Note: I've labeled the input directly via the legend too. That way it has a label announced each time it is focused. When a user focuses the first button, for the first time, they would hear "How many infants, under two years old, are traveling? button, increment")

**Comment by adamsilver** [comment]
1. Thinking aloud: how about we remove verbosity by putting aria-labelledby/describedby on the button instead of aria-label?
2. Discuss live region overkill

**Comment by Heydon** [comment]
@adamsilver Looks like I forgot the live region in the example...

```html
<fieldset>
    <legend id="under-two">How many infants, under two years old, are traveling?</legend>
    <button type="button" aria-label="Decrement">&minus;</button>
    <input type="number" id="adults" name="adults" aria-labelledby="under-two" role="status">
    <button type="button" aria-label="Increment">&plus;</button>
</fieldset>
```

**Comment by adamsilver** [comment]
Ran a quick test:

In JAWS/IE9: The legend is announced twice when focused on the text box. On both buttons it says “LEGEND, increment, button”
In NVDA/FF: The group label isn't announced on the buttons
In Safari/Voiceover: The first time you enter the group, the button is announced along with the legend. After that it's not announced

**Comment by adamsilver** [comment]
2. could do fake fieldset and aria-labelled/describedby.

**Comment by adamsilver** [comment]
Or label the increment button by the label (give the label an id).

**Comment by adamsilver** [comment]
Note: one other reason I don't like the fieldset here, is that it's not clickable like a label would be.

**Comment by adamsilver** [comment]
# Testing the input as a live region

Doing this so don't have to create a separate element. Explain why not doing it in the text box. Live region not necessarily crucial. Etc.

## Chrome + voiceover 
- no live region: hitting increment will announce the change from 0 to 1. But not 1 to 2.
- input live region: same as no live region

## Safari + voiceover
- no live region: hitting increment announces nothing.
- input live region: hitting increment announces the input's value change (Good)

## IE11 + JAWS
- no live region: hitting increment announces nothing.
- input live region: same as when no live region

## FF + NVDA
- no live region: hitting increment announces nothing.
- input live region: announces the change (good) but when focusing the number input announces it as a status (“<label>, status, edit”) (BAD). 

# Testing aria-label (to change + to “increment”) and aria-describedby using the <label> to describe the button

## Safari + voiceover
- announces both aria-label and describedby as standard (reveal help tag via CTRL+OPTION+SHIFT+H) (good)

## FF + NVDA
- announces “increment button, how many adults are flying...” (good)

## IE + JAWS
- announces “increment button, how many adults are flying...” (good)
